### PR TITLE
Init

### PR DESCRIPTION
**Pull request by delvedor**
Initial implementation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,153 @@
...
+
+upring.use(require('upring-kv'))
+upring.use(require('upring-set'))
+
+upring.on('up', onReady)
+
+async function onReady () {
+  await upring.set.sadd('set', ['one', 'two', 'three'])
+  console.log(upring.set.smembers('set'))
+}
```
Comment (mcollina) [inline README.md]: this is very very ugly to read. ee and promises... brrr

File: index.js [hunk 2]
```diff
@@ -0,0 +1,293 @@
+'use strict'
+
+require('make-promises-safe')
+
+const _ = require('lodash')
```
Comment (mcollina) [inline index.js]: I would not use _ if possible, not a fan.

File: index.js [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+const assert = require('assert')
+const deepEqual = require('fast-deep-equal')
+
+function setPlugin (upring, opts, next) {
+  if (upring.kv === undefined) {
+    return next(new Error('Missig kv plugin'))
+  }
+
+  upring.set = new SetUtils(upring, opts)
+  next()
```
Comment (mcollina) [inline index.js]: do this need to be a callback? we are using async/await everywhere.

**Review by mcollina — APPROVED** [review]
LGTM

File: index.js (hunk 2 above)
Comment (delvedor) [reply index.js]: Not a fan of lodash as a library or the character "_"?

**Pull request merged**